compiler/rustc_index/src/bit_set.rs

```diff
@@ -743,6 +787,38 @@ impl<'a, T: Idx> Iterator for ChunkedBitIter<'a, T> {
         }
         None
     }
+
+    fn fold<B, F>(mut self, mut init: B, mut f: F) -> B
+    where
+        F: FnMut(B, Self::Item) -> B,
+    {
+        while self.index % CHUNK_BITS != 0 {
```

Comment:
It would be good to have a comment here explaining how the `while` loop moves to the start of a chunk (necessary if we're not starting at the start of the bitset), and the rest goes from there in chunk-sized steps.